com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/cf/metadata/entity/processor/MtaMetadataEntityAggregator.java

```diff
@@ -0,0 +1,106 @@
+package com.sap.cloud.lm.sl.cf.core.cf.metadata.entity.processor;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.cloudfoundry.client.lib.domain.CloudApplication;
+import org.cloudfoundry.client.lib.domain.CloudEntity;
+import org.cloudfoundry.client.lib.domain.CloudService;
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.ImmutableMtaMetadata;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.MtaMetadata;
+import com.sap.cloud.lm.sl.cf.core.cf.metadata.processor.MtaMetadataParser;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMta;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaModule;
+import com.sap.cloud.lm.sl.cf.core.model.DeployedMtaResource;
+import com.sap.cloud.lm.sl.cf.core.model.ImmutableDeployedMta;
+import com.sap.cloud.lm.sl.mta.model.Version;
+
+@Component
+public class MtaMetadataEntityAggregator {
+
+    @Inject
+    private MtaMetadataParser mtaMetadataParser;
+
+    public List<DeployedMta> aggregate(List<CloudEntity> entities) {
+        Map<String, List<CloudEntity>> entitiesByMtaId = getEntitiesByMtaId(entities);
+        return entitiesByMtaId.values()
+                              .stream()
+                              .map(this::toDeployedMta)
+                              .collect(Collectors.toList());
+    }
+
+    private Map<String, List<CloudEntity>> getEntitiesByMtaId(List<CloudEntity> entities) {
```

Comment:
can be substituted with
`entities.stream().collect(groupingBy(entity -> parseMetadata(entity).getId())`

Reply:
Same as above response.